server/server.go

```diff
@@ -47,6 +69,81 @@ func (cs *cribbageServer) Serve() {
 	}
 }
 
+func (cs *cribbageServer) ginPostRegister(c *gin.Context) {
+	u, p, err := decodeUserAndPass(c)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	creds, err := auth.NewCredentials(u, p)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	player := model.Player{
+		Credentials: creds,
+		ID:          model.PlayerID(creds.Username),
+		Name:        `player`,
+	}
+	err = createPlayer(context.Background(), player)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	// TODO make the key more secret (environment var), change expiration duration to much shorter
+	jwtSvc := auth.NewJWTService(`somethingSecret`, time.Minute*180)
+	cookie, err := jwtSvc.CreateCookie(creds.Username)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	c.SetCookie(cookie.Name, cookie.Value, cookie.MaxAge, cookie.Path, cookie.Domain, cookie.Secure, cookie.HttpOnly)
+}
+
+func (cs *cribbageServer) ginPostLogin(c *gin.Context) {
+	u, p, err := decodeUserAndPass(c)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	player, err := getPlayer(model.PlayerID(u))
```

Comment:
how does this load up a user's password from persistence so that `ValidateCredentials` later can verify it?

Reply:
`Credentials` is embedded in `Player`, so the DB abstractions should handle loading that up properly (I checked the mongo implementation and it seems to work)